polyval/src/field.rs

```diff
@@ -14,10 +14,23 @@
 //!
 //! [RFC 8452 Section 3]: https://tools.ietf.org/html/rfc8452#section-3
 
-pub mod backend;
+#[cfg(all(
+    target_feature = "pclmulqdq",
+    target_feature = "sse2",
+    target_feature = "sse4.1",
+    any(target_arch = "x86", target_arch = "x86_64")
+))]
```

Comment:
The conditional gating this implementation is using is gross. Open to alternatives... perhaps the [`cfg-if` crate](https://crates.io/crates/cfg_if)?

Reply:
I tried `cfg-if`. Unfortunately it doesn't seem to support nesting, and the resulting code ended up looking just as bad

Reply:
Made things a little better by using `cfg!`. Still a lot of annoying, redundant gating though

Reply:
My kingdom for `cfg_alias`: https://github.com/rust-lang/cargo/issues/7260#issuecomment-532953766